.changeset/tall-comics-compare.md

```diff
@@ -0,0 +1,9 @@
+---
+"@itwin/presentation-hierarchies": minor
+---
+
+- Increased the speed of hierarchy filtering with large number of filtered paths.
+
+- Added `compare` function to `HierarchyNodeIdentifier`.
+
+- Changed `NodeParser`, it now returns `SourceInstanceHierarchyNode | Promise<SourceInstanceHierarchyNode>`.
```

Comment:
The first row should not be a bullet point, otherwise the final changelog renders weirdly. Also made API addition descriptions more specific.

```suggestion
Increased the speed of hierarchy filtering with large number of filtered paths.

- Added `HierarchyNodeIdentifier.compare` function to compare two identifiers.
- Changed `NodeParser` (return type of `HierarchyDefinition.parseNode`), to additionally allow returning a promise, so instead of just `SourceInstanceHierarchyNode` it can now also return `Promise<SourceInstanceHierarchyNode>`.
```